internal/resources/resource_host.go

```diff
@@ -776,31 +785,49 @@ func getNetworkIDs(d *schema.ResourceData, p *configuration.Config, host *rest.H
 	return netIds, nil
 }
 
-// getNetworkDefRoute returns the network for default route specified in the request or
-// the current value maintained in the state.
-func getNetworkDefRoute(d *schema.ResourceData, p *configuration.Config, host *rest.Host) (nDefRoute string, err error) {
-	nDefRoute, ok := d.Get(hNetForDefaultRoute).(string)
-	if !ok {
-		// network for default route is optional
-		// return the current value
-		return host.NetworkForDefaultRoute, nil
+// getNetworkID returns the network ID specified in the request.
+func getNetworkID(p *configuration.Config, hostNets []string, locationID, net string) (string, error) {
+	if net == "" {
+		return net, nil
 	}
 
-	nIDMap, nNameMap := getAvailableNetworkMaps(p, host.LocationID)
+	nIDMap, nNameMap := getAvailableNetworkMaps(p, locationID)
 	if len(nIDMap) == 0 {
-		return nDefRoute, fmt.Errorf("no available networks for location %s", host.LocationID)
+		return "", fmt.Errorf("no available networks for location %s", locationID)
 	}
 
-	if _, ok := nIDMap[nDefRoute]; ok {
-		return nDefRoute, nil
+	var (
+		found   bool
+		network string
+	)
+
+	if _, ok := nIDMap[net]; ok {
+		for _, netID := range hostNets {
+			if netID == net {
+				found = true
+				network = net
+
+				break
+			}
+		}
+	} else if id, ok := nNameMap[net]; ok {
+		for _, netID := range hostNets {
+			if netID == id {
+				found = true
+				network = id
+
+				break
+			}
+		}
+	} else {
+		return "", fmt.Errorf("network %s does not match any available network for location %s", net, locationID)
```

Comment:
This complex if looks like it might be better as a switch? @mchuang3

Reply:
> This complex if looks like it might be better as a switch? @mchuang3

This is probably ok, as there's an implied order preference to check for ID first, then Name.  Switches are in theory random order.  But probably doesn't matter either way on this.  Since it's existing code that got moved, I'd just leave it as is.